### PR TITLE
feat: policypanda script to help check if an orgs repos follows requirements/recommendations.

### PR DESCRIPTION
**Pull request by maxandersen**
quick'n'dirty best-effort on a script that can be used to check if a github's org repos follows the guidelines.

It relies on GitHub APIs for now to avoid requiring local clones of all the content. Once we have such projects, it could be evolved to support GitHub, etc

for now assume dco style project by default but can use `-t cla` to assume cla instead.

Example output:

```
## Policy Panda 🐼
The Policy Panda tries to spot if org/repos follows the requirement and recommendations by the CommonHaus Foundation.
It will look for the various files in various possbile formats ([.md, .adoc, .txt]) in the repositories, including `.github` global repo if present.
It is not perfect, it is just a panda. 🐼
If you think it is missing something, please let us know at https://github.com/commonhaus/foundation/issues
## commonhaus/foundation
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/foundation/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/foundation/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/foundation/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/foundation/blob/main/CODE_OF_CONDUCT.md)
## commonhaus/.github
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/.github/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/.github/blob/main/CODE_OF_CONDUCT.md)
## commonhaus/commonhaus.github.io
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/.github/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/commonhaus.github.io/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/commonhaus.github.io/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/.github/blob/main/CODE_OF_CONDUCT.md)
## commonhaus/foundation-internal
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/.github/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/.github/blob/main/CODE_OF_CONDUCT.md)
## commonhaus/automation
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/.github/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/automation/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/automation/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/.github/blob/main/CODE_OF_CONDUCT.md)
## commonhaus/artwork
- ⚠️ Code owners - Code owners file
- ✅ Governance - Governance file (commonhaus/.github/blob/main/GOVERNANCE.md)
- ❌ DCO - DCO file should be present
- ✅ Contributing guide - Contributing guide present (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ⚠️ DCO - Contributing guide should mention DCO (commonhaus/.github/blob/main/CONTRIBUTING.md)
- ✅ Code of conduct - Code of conduct file (commonhaus/.github/blob/main/CODE_OF_CONDUCT.md)
```

thinking of adding additional check if the files contains the suggested content or has a variant; but for now just doing the basic presence detection.

### CODE REVIEW

**Comment by maxandersen** [comment]
note, it is more "friendly" than the policies are written by looking both in repo and in <org>/.github  + accepting .txt,.md and .adoc for file formats and also allowing lower case names (dco.txt vs DCO.txt as that's what hibernate uses)

**Review by kenfinnigan — APPROVED** [review]
Would it be something we set it up to run as a CI job when a new project application PR is received?

To have a sense on the PR of what pieces we require of a project that aren't already there.

File: templates/README.md [hunk 1]
```diff
@@ -36,3 +36,19 @@ The files in this folder are our recommendations for what CF projects should def
 - **CODEOWNERS** to define individuals or teams that are responsible for code in a repository
     - [About Code Owners (GitHub)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners).
     - [Code Owners (GitLab)](https://docs.gitlab.com/ee/user/project/codeowners/)
+
+## Policy Panda
+
+This repository has a script that does a basic check if a GitHub repository/organization has the recommended files.
```
Comment (kenfinnigan) [inline templates/README.md:42]: Can you remove the trailing space on this line

File: templates/README.md [hunk 2]
```diff
@@ -36,3 +36,19 @@ The files in this folder are our recommendations for what CF projects should def
...
+
+You run it like this: `jbang policypanda@commonhaus/foundation hibernate` where `hibernate` is the GitHub organization name.
+
+If you want it to only check specific repositories, you can use the `-r` to specify a regular expression.
+
+Example:
+
+```shell
+jbang policypanda@commonhaus/foundation hibernate -r "hibernate-orm|hibernate-validator"
+```
```
Comment (kenfinnigan) [inline templates/README.md:52]: Trailing space here too

File: templates/README.md [hunk 3]
```diff
@@ -36,3 +36,19 @@ The files in this folder are our recommendations for what CF projects should def
...
+
+If you want it to only check specific repositories, you can use the `-r` to specify a regular expression.
+
+Example:
+
+```shell
+jbang policypanda@commonhaus/foundation hibernate -r "hibernate-orm|hibernate-validator"
+``` 
+
+would only check the repositories that match `hibernate-orm` or `hibernate-validator`.
```
Comment (kenfinnigan) [inline templates/README.md:54]: Linting requires a blank line at the end of the file

File: templates/README.md (hunk 1 above)
Comment (ebullient) [inline templates/README.md:42]:
```suggestion
This repository has a script that does a basic check if a GitHub repository/organization has the recommended files.
```

File: templates/README.md (hunk 2 above)
Comment (ebullient) [inline templates/README.md:52]:
````suggestion
```
````

File: templates/README.md (hunk 3 above)
Comment (ebullient) [inline templates/README.md:54]:
```suggestion
would only check the repositories that match `hibernate-orm` or `hibernate-validator`.

```

**Pull request merged**